samples/03_sharded_client/Services/CommandHandlingService.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+
+namespace _03_sharded_client.Services
+{
+    public class CommandHandlingService
+    {
+        private readonly CommandService _commands;
+        private readonly DiscordShardedClient _discord;
+        private readonly IServiceProvider _services;
+
+        public CommandHandlingService(IServiceProvider services)
+        {
+            _commands = services.GetRequiredService<CommandService>();
+            _discord = services.GetRequiredService<DiscordShardedClient>();
+            _services = services;
+
+            _discord.MessageReceived += MessageReceivedAsync;
```

Comment:
I'd probably move it under `InitializeAsync` so it doesn't start processing before modules are discovered.

Reply:
"It" just being the handler attachment?

Reply:
Yes.